services.py

```diff
@@ -12,3 +12,10 @@ def init_model(self, module_str):
         assert model is not None
 
         self.model = model
+
+    def init_models(self, module_strs):
+        self.models = {}
```

Comment:
Interesting. I don't think we need to preserve the original mapping, because it can always be derived some other way. Just make it a list.

```
self.models = [resolve_model_dynamically(module_str) for module_str in module_strs]
assert all(self.models)
```

Will check that all are truthy (none are falsy, or `None`)